nifi-nar-bundles/nifi-cdc/nifi-cdc-mssql-bundle/nifi-cdc-mssql-processors/src/test/java/org/apache/nifi/cdc/mssql/CaptureChangeMSSQLTest.java

```diff
@@ -0,0 +1,792 @@
+
+        runner.getStateManager().clear(Scope.CLUSTER);
+    }
+
+    @After
+    public void teardown() throws IOException {
+        runner.getStateManager().clear(Scope.CLUSTER);
+        runner = null;
+    }
+
+    @Test
+    public void testSelectGenerator(){
+        MSSQLCDCUtils mssqlcdcUtils = new MSSQLCDCUtils();
+
+        List<ColumnDefinition> columns = new ArrayList<>();
+        columns.add(new MSSQLColumnDefinition(Types.INTEGER, "ID", 1, true));
+        columns.add(new MSSQLColumnDefinition(Types.VARCHAR, "LastName", 2, false));
+        columns.add(new MSSQLColumnDefinition(Types.VARCHAR, "FirstName", 3, false));
+
+        MSSQLTableInfo ti = new MSSQLTableInfo("NiFi", "cdc", "Names",
+                "dbo", "dbo_Names_CT", 1000L, columns);
+
+        String noMaxTime = mssqlcdcUtils.getCDCSelectStatement(ti, null);
+
+        Assert.assertEquals("SELECT t.tran_begin_time\n" +
+                ",t.tran_end_time \"tran_end_time\"\n" +
+                ",CAST(t.tran_id AS bigint) trans_id\n" +
+                ",CAST(\"o\".\"__$start_lsn\" AS bigint) start_lsn\n" +
+                ",CAST(\"o\".\"__$seqval\" AS bigint) seqval\n" +
+                ",\"o\".\"__$operation\" operation\n" +
+                ",CAST(\"o\".\"__$update_mask\" AS bigint) update_mask\n" +
+                ",\"o\".\"ID\"\n" +
+                ",\"o\".\"LastName\"\n" +
+                ",\"o\".\"FirstName\"\n" +
+                ",CURRENT_TIMESTAMP EXTRACT_TIME\n" +
+                "FROM cdc.\"Names\" \"o\"\n" +
+                "INNER JOIN cdc.lsn_time_mapping t ON (t.start_lsn = \"o\".\"__$start_lsn\")\n" +
+                "ORDER BY CAST(\"o\".\"__$start_lsn\" AS bigint), \"o\".\"__$seqval\", \"o\".\"__$operation\"", noMaxTime);
+
+        String withMaxTime = mssqlcdcUtils.getCDCSelectStatement(ti, new Timestamp(0));
+
+        Assert.assertEquals("SELECT t.tran_begin_time\n" +
+                ",t.tran_end_time \"tran_end_time\"\n" +
+                ",CAST(t.tran_id AS bigint) trans_id\n" +
+                ",CAST(\"o\".\"__$start_lsn\" AS bigint) start_lsn\n" +
+                ",CAST(\"o\".\"__$seqval\" AS bigint) seqval\n" +
+                ",\"o\".\"__$operation\" operation\n" +
+                ",CAST(\"o\".\"__$update_mask\" AS bigint) update_mask\n" +
+                ",\"o\".\"ID\"\n" +
+                ",\"o\".\"LastName\"\n" +
+                ",\"o\".\"FirstName\"\n" +
+                ",CURRENT_TIMESTAMP EXTRACT_TIME\n" +
+                "FROM cdc.\"Names\" \"o\"\n" +
+                "INNER JOIN cdc.lsn_time_mapping t ON (t.start_lsn = \"o\".\"__$start_lsn\")\n" +
+                "WHERE t.tran_end_time > ?\n" +
+                "ORDER BY CAST(\"o\".\"__$start_lsn\" AS bigint), \"o\".\"__$seqval\", \"o\".\"__$operation\"", withMaxTime);
+    }
+
+    @Test
+    public void testRetrieveAllChanges() throws SQLException, IOException {
+        setupNamesTable();
+        
```

Comment:
Line 215 is not blank, so it throws a CheckStyle violation

Reply:
I've never been able to get CheckStyle to run on my unit tests. It's like IntelliJ doesn't recognize them as valid targets for the plugin. Thanks.